lib/spack/spack/patch.py

```diff
@@ -530,7 +530,7 @@ def update_package(self, pkg_fullname: str) -> None:
 
         # update the index with per-package patch indexes
         pkg_cls = self.repository.get_pkg_class(pkg_fullname)
-        partial_index = self._index_patches(pkg_cls, self.repository)
+        partial_index = self._index_patches(pkg_cls)
```

Comment:
@tgamblin @becker33 Note that this and the line above:
```python
pkg_cls = self.repository.get_pkg_class(pkg_fullname)
partial_index = self._index_patches(pkg_cls)
```
was the main cause for my request :point_up: Now we are using both  `self.repository` and `spack.repo.PATH`. We should keep that in mind, because if they ever get out of sync (and they probably do if this fixes the issue), they will cause bugs.

Reply:
What would be out of sync here?  In any given Spack run, the dependency a name refers to is the one in `spack.repo.PATH`.  So this is the only way I see right now to resolve a name that's not in the local repo, and it's the only one we really support (unless the dependency explicitly specifies a namespace, but then you want to go through `PATH` as well).

I'm not sure what you want this to do.